homeassistant/components/life360/__init__.py

```diff
@@ -28,184 +22,138 @@
+    # update with missing data.
+    if not entry.unique_id:
+        hass.config_entries.async_update_entry(
+            entry,
+            unique_id=entry.data[CONF_USERNAME].lower(),
+            options=DEFAULT_OPTIONS | hass.data[DOMAIN].cfg_options,
+        )
 
-CONFIG_SCHEMA = vol.Schema({DOMAIN: LIFE360_SCHEMA}, extra=vol.ALLOW_EXTRA)
+    api = get_life360_api(authorization=entry.data[CONF_AUTHORIZATION])
 
+    async def async_update_data() -> Life360Data:
+        """Update Life360 data."""
+        return await get_life360_data(hass, api)
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up integration."""
-    conf = config.get(DOMAIN, LIFE360_SCHEMA({}))
-    hass.data[DOMAIN] = {"config": conf, "apis": {}}
-    discovery.load_platform(hass, Platform.DEVICE_TRACKER, DOMAIN, None, config)
-
-    if CONF_ACCOUNTS not in conf:
-        return True
-
-    # Check existing config entries. For any that correspond to an entry in
-    # configuration.yaml, and whose password has not changed, nothing needs to
-    # be done with that config entry or that account from configuration.yaml.
-    # But if the config entry was created by import and the account no longer
-    # exists in configuration.yaml, or if the password has changed, then delete
-    # that out-of-date config entry.
-    already_configured = []
-    for entry in hass.config_entries.async_entries(DOMAIN):
-        # Find corresponding configuration.yaml entry and its password.
-        password = None
-        for account in conf[CONF_ACCOUNTS]:
-            if account[CONF_USERNAME] == entry.data[CONF_USERNAME]:
-                password = account[CONF_PASSWORD]
-        if password == entry.data[CONF_PASSWORD]:
-            already_configured.append(entry.data[CONF_USERNAME])
-            continue
-        if (
-            not password
-            and entry.source == config_entries.SOURCE_IMPORT
-            or password
-            and password != entry.data[CONF_PASSWORD]
-        ):
-            hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
-
-    # Create config entries for accounts listed in configuration.
-    for account in conf[CONF_ACCOUNTS]:
-        if account[CONF_USERNAME] not in already_configured:
-            hass.async_create_task(
-                hass.config_entries.flow.async_init(
-                    DOMAIN,
-                    context={"source": config_entries.SOURCE_IMPORT},
-                    data=account,
-                )
-            )
-    return True
+    if not (coordinator := hass.data[DOMAIN].coordinators.get(entry.unique_id)):
+        coordinator = DataUpdateCoordinator(
+            hass,
+            LOGGER,
+            name=f"{DOMAIN} ({entry.unique_id})",
+            update_interval=_update_interval(entry),
+            update_method=async_update_data,
+        )
+        hass.data[DOMAIN].coordinators[entry.unique_id] = coordinator
```

Comment:
Please use the config entry entry_id as key to store things in hass.data. And don't store things in hass.data until we know we won't raise ConfigEntryNotReady, ie after the first coordinator refresh.

Reply:
Ok.